core/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramParser.java

```diff
@@ -52,6 +51,8 @@ protected DateHistogramAggregationBuilder createFactory(String aggregationName,
             factory.interval((Long) interval);
         } else if (interval instanceof DateHistogramInterval) {
             factory.dateHistogramInterval((DateHistogramInterval) interval);
+        } else {
+            throw new AssertionError();
```

Comment:
Feel free to tell me I'm totally wrong - but do we feel confident enough that a mistake here (in code) is a good enough reason to cause the [node to restart](https://github.com/elastic/elasticsearch/pull/19272) ? I'm worried about endless restarts, where a single mistake on a specific API can cause problems on an entire node.